### PR TITLE
Bug 1922992: i18n missing items in YAML sidebar

### PR DESCRIPTION
**Pull request by rebeccaalpert**
Resource kind labels and button text were not internationalized. I updated this. The rest of the schema details isn't available for internationalization at this time unless we hard-code it.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1922992.

### CODE REVIEW

File: frontend/public/components/sidebars/explore-type-sidebar.tsx [hunk 1]
```diff
@@ -49,8 +52,9 @@ export const ExploreType: React.FC<ExploreTypeProps> = (props) => {
     ? currentSelection.description
     : currentDefinition.description;
   const required = new Set(currentDefinition.required || []);
+  const kindLabel = kindObj.labelKey ? t(kindObj.labelKey) : kindObj.kind;
   const breadcrumbs = drilldownHistory.length
-    ? [kindObj ? kindObj.kind : 'Schema', ..._.map(drilldownHistory, 'name')]
+    ? [kindObj ? kindLabel : t('public~Schema'), ..._.map(drilldownHistory, 'name')]
```
Comment (spadgett) [inline frontend/public/components/sidebars/explore-type-sidebar.tsx:57]: We check if `kindObj` is set above (line 31), so I don't think this fallback is needed.

File: frontend/public/components/sidebars/resource-sidebar-samples.tsx [hunk 2]
```diff
@@ -97,7 +101,7 @@ const ResourceSidebarSnippet: React.FC<ResourceSidebarSnippetProps> = ({
       <p>{description}</p>
       <Button type="button" variant="link" isInline onClick={insertSnippet}>
         <PasteIcon className="co-icon-space-r" />
-        Insert Snippet
+        {t('public~Insert Snippet')}
```
Comment (spadgett) [inline frontend/public/components/sidebars/resource-sidebar-samples.tsx:104]:
```suggestion
        {t('public~Insert snippet')}
```

File: frontend/public/components/sidebars/resource-sidebar.tsx [hunk 3]
```diff
@@ -74,7 +74,8 @@ export const ResourceSidebar = (props) => {
     return null;
   }
 
-  const { label = sidebarLabel } = kindObj ?? {};
+  const kindLabel = kindObj?.labelKey ? t(kindObj.labelKey) : kindObj?.label;
+  const label = sidebarLabel ? sidebarLabel : kindLabel ?? {};
```
Comment (spadgett) [inline frontend/public/components/sidebars/resource-sidebar.tsx:78]: Why are we falling back to `{}` when the value should be a string?

**Review by spadgett — CHANGES_REQUESTED** [review]

File: frontend/public/components/sidebars/explore-type-sidebar.tsx (hunk 1 above)
Comment (rebeccaalpert) [reply frontend/public/components/sidebars/explore-type-sidebar.tsx:57]: You can have no kindObj as long as you have a schema, so you do need the fallback.

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**